cmd/snap/cmd_run.go

```diff
@@ -133,7 +133,15 @@ func createUserDataDirs(info *snap.Info) error {
 			return fmt.Errorf(i18n.G("cannot create %q: %v"), d, err)
 		}
 	}
-	return nil
+
+	// 'current' symlink for user data (SNAP_USER_DATA)
+	currentActiveSymlink := filepath.Join(userData, "..", "current")
+	if err := os.Remove(currentActiveSymlink); err != nil && !os.IsNotExist(err) {
```

Comment:
We should not remove the symlink as that may randomly break applications that concurrently 1) keep running and use it 2) try to create it.

I think you really have to:
- acquire an exclusive `flock(2)` based lock on `/home/snap/$SNAP_NAME/.lock
- `readlink` the symlink file
  - if that fails because there's no symlink then:
      - remember that we want to create it
      - ignore the error
  - if that fails because the symlink file is not a symlink then:
      - exit with an error that is sensible to the user
  - if that works ok then check the value, if it is incorrect then:
    - remove the symlink file
    - remember that we want to create it
- if we want to create the symlink then create it with the right value
- unlock the lock file